docs/guides/implement_custom_components.md

```diff
@@ -0,0 +1,171 @@
+# Guide - Implement custom components
+
+**Level**: Beginner
+**Time**: 20min
+**Goal**: After completing this tutorial with Fondant, you will be able to build your own custom component and integrate it into a fondant pipeline. 
+
+**Prerequisite**: Make sure docker compose is installed on your local system.
+We recommend completing the [first tutorial](/docs/guides/build_a_simple_pipeline.md) before proceeding with this one, as this tutorial builds upon the knowledge gained in the previous one.
+
+## Overview
+
+In the [initial tutorial](/docs/guides/build_a_simple_pipeline.md), you learned how to create your first Fondant pipeline. While the example demonstrates initialising the dataset from HuggingFace and using a reusable component to download images, this is just the beginning.
+
+The true power of Fondant lies in its capability to enable you to construct your own data pipelines to create high-quality datasets. To achieve this, we need to implement custom components.
+
+In this tutorial, we will guide you through the process of implementing your very own custom component. We will illustrate this by demonstrating how to filter images based on file type.
+
+This pipeline is an extension of the one introduced in the first tutorial. After loading the dataset from HuggingFace, it narrows down the dataset to exclusively encompass PNG files. Finally, it proceeds to download these carefully selected images.
+
+
+## Build your custom component
+
+A component comprises several key elements. First, there's the ComponentSpec YAML file, serving as a blueprint for the component. It defines crucial aspects such as input and output dataframes, along with component arguments.
+
+![component architecture](/docs/art/guides/component.png)
+
+The second essential part is a python class, which encapsulates the business logic that operates on the input dataframe.
+
+In addition to these core components, there are a few other necessary items, including a `Dockerfile` used for building the component and a `requirements.txt` file to specify and install required dependencies. You can find a more detailed explanation [here](custom_components.md).
+
+### Creating the ComponentSpec
+
+First of all we create the following ComponentSpec (fondant_component.yaml) file in the folder `components/filter_images`: 
+
+```yaml
+name: Filter file type
+description: Component that filters on mime types
+image: filter_images
```

Comment:
This should be a url it's not really just an image name

Reply:
That was the reason why I decided here for `filter_image`. Technical, it is really needed if we are using the LocalRunner. Any placeholder would work. Shall we create an issue to loosen up the validation of the spec a bit?

Reply:
In the `LocalRunner` it dictates the name of the docker image that's built. We can keep it as `filter_images` for now since we're just running it locally.